### ISSUE TITLE
Add config parameters to network constructors: n_channel_input and output_softmax?

### ISSUE THREAD

**Issue by jonalm** [FluxML/Metalhead.jl#95]
Hi there,

I'm trying out the image networks for classification of data with some imagy-properities (e.g. x-y-locality) but not 3 channels like conventional images. As far as I can tell, the 3 input channels are hard coded in the network implementation. Would you consider a PR adding a `n_channel_input = 3` keyword argument to the networks?

Also, would you consider a PR for an optional `output_softmax = false` parameter, which (if true) adds a final softmax layer?

best
Jon

**Comment by darsnack** [comment]
Just to clarify, are you working off the last release or the master branch?

**Comment by jonalm** [comment]
Master

**Comment by darsnack** [comment]
For accepting multiple input channels, that's definitely something we'd like to support. The VGG models already accept such a keyword. For something like ResNet, there is usually an initial conv-batchnorm-activation before all the residual blocks. There, I think it would be nice to accept any Flux model as the "initial backbone."

For the softmax layer, I feel like the cleaner design is to just tack on a softmax at the end. Something like
```julia
model = Chain(ResNet50(), softmax)
```
Generally, we try to limit the number of keywords to keep the API simple.

**Comment by DhairyaLGandhi** [comment]
Right, it seems like we should not have `softmax` in the models by default. I even wonder if we should make the classification head optional. In most cases, that would be dropped and retrained anyway, but it is good to have it around optionally when its convenient not to have to do that (say in demos). Ditto for `bias` and `activation`.

**Comment by darsnack** [comment]
I'm going to close this as a duplicate of #176 which has a more up to date discussion.
